File: ios/AwesomeProject/Module/EventEmitterModule.m
//
//  EventEmitterModule.m
//  AwesomeProject
//
//  Created by 林 on 2024/7/27.
//

#import "EventEmitterModule.h"


@implementation EventEmitterModule

RCT_EXPORT_MODULE();

- (NSArray<NSString *> *)supportedEvents {
  return @[@"UsersLoaded"];
}

RCT_EXPORT_METHOD(emitEvent:(NSString *)name body:(NSDictionary *)body) {
  [self sendEventWithName:name body:body];
}

+ (BOOL)requiresMainQueueSetup {
  return YES;
}

@end

File: ios/AwesomeProject/Module/EventEmitterModule.swift
//
//  EventEmitterModule.swift
//  AwesomeProject
//
//  Created by 林 on 2024/7/22.
//

import Foundation
import Foundation
import React

@objc(EventEmitterModule)
class EventEmitterModule: RCTEventEmitter {
  override init() {
    super.init()
    NotificationCenter.default.addObserver(self, selector: #selector(emitEventInternal(_:)), name: Notification.Name("EventEmitterModuleEvent"), object: nil)
  }

  override static func requiresMainQueueSetup() -> Bool {
    return true
  }

  @objc override func supportedEvents() -> [String] {
    return ["UsersLoaded"]
  }

  @objc func emitEventInternal(_ notification: Notification) {
    if let userInfo = notification.userInfo, let name = notification.name.rawValue as String? {
      sendEvent(withName: name, body: userInfo)
    }
  }
  
    @objc func emitEvent(_ name: String, body: [String: Any]) {
        sendEvent(withName: name, body: body)
    }
}
